subprojects/core/src/main/java/org/gradle/internal/buildtree/DefaultBuildTreeLifecycleController.java

```diff
@@ -72,7 +78,7 @@ public void scheduleAndRunTasks(EntryTaskSelector selector) {
     public <T> T fromBuildModel(boolean runTasks, BuildTreeModelAction<? extends T> action) {
         return runBuild(() -> {
             modelCreator.beforeTasks(action);
-            if (runTasks) {
+            if (runTasks && isEligibleToRunTasks()) {
```

Comment:
Maybe add a comment that this is a temporary workaround.

It would also be good to somehow connect this to `Workarounds` so that it can remind us to remove the workaround in the future.